src/community/ardrive_contract_oracle.ts

```diff
@@ -0,0 +1,155 @@
+import Arweave from 'arweave';
+import { ContractOracle, ContractReader } from './contract_oracle';
+import { VertoContractReader } from './verto_contract_oracle';
+import { SmartweaveContractReader } from './smartweave_contract_oracle';
+// import { RedstoneContractReader } from './redstone_contract_oracle';
+
+/* eslint-disable no-console */
+
+// ArDrive Profit Sharing Community Smart Contract
+export const communityTxId = '-8A6RexFkpfWwuyVO98wzSFZh0d6VJuI-buTJvlwOJQ';
+
+const maxReadContractAttempts = 3;
+const initialContractReader = 0;
+const initialContractAttempts = 0;
+
+export interface CommunityContractData {
+	votes: [Record<string, unknown>];
+	settings: [string, unknown][];
+	balances: { [tokenHolderAddress: string]: number };
+	vault: { [tokenHolderAddress: string]: { balance: number; start: number; end: number }[] };
+}
+
+/**
+ * Oracle class responsible for retrieving the correct data fields from
+ * the ArDrive Community Contract. This class can utilize several different
+ * contract readers and will fallback to other readers if one fails
+ *
+ * @remarks Will begin fetching data from default contract reader on construction
+ */
+export class ArDriveContractOracle implements ContractOracle {
+	constructor(private readonly arweave: Arweave) {
```

Comment:
consider a `skipSetup` flag for unit testing purposes.

Reply:
Might add more complexity than we want for this ticket right now so a TODO is fine.